tests/tests/test-precompile-erc20.ts

```diff
@@ -0,0 +1,258 @@
+import { expect } from "chai";
+import { describeDevMoonbeam } from "../util/setup-dev-tests";
+import { customWeb3Request } from "../util/providers";
+import {
+  GENESIS_ACCOUNT,
+  ALITH,
+  BALTATHAR,
+  ALITH_PRIV_KEY,
+  CHARLETH,
+  BALTATHAR_PRIV_KEY,
+} from "../util/constants";
+import { createTransaction } from "../util/transactions";
+
+const ADDRESS_ERC20 = "0x0000000000000000000000000000000000000802";
+const SELECTORS = {
+  balanceOf: "70a08231",
+  totalSupply: "18160ddd",
+  approve: "095ea7b3",
+  allowance: "dd62ed3e",
+  transfer: "a9059cbb",
+  transferFrom: "23b872dd",
+  logApprove: "0x8c5be1e5ebec7d5bd14f71427d1e84f3dd0314c0f7b2291e5b200ac8c7c3b925",
+  logTransfer: "0xddf252ad1be2c89b69c2b068fc378daa952ba7f163c4a11628f55a4df523b3ef",
+};
+const GAS_PRICE = "0x" + (1_000_000_000).toString(16);
+
+async function getBalance(context, blockHeight, address) {
+  const blockHash = await context.polkadotApi.rpc.chain.getBlockHash(blockHeight);
+  const account = await context.polkadotApi.query.system.account.at(blockHash, address);
```

Comment:
Note: you can call `const account = await context.polkadotApi.query.system.account(address)` to get the balance at current block